src/lib/components/layouts/DocumentBrowser.svelte

```diff
@@ -74,6 +80,26 @@
     HIDE_COUNT: width < remToPx(26),
   };
 
+  $: searchResults = documents.then((r) => excludeDeleted(r.data, $deleted));
+
+  // filter out deleted documents that haven't been purged from search yet
+  function excludeDeleted(
+    searchResults: DocumentResults,
+    deleted: Set<string>,
+  ): DocumentResults {
+    if (deleted.size === 0) return searchResults;
+
+    const filtered = searchResults.results.filter(
+      (d) => !deleted.has(String(d.id)),
+    );
```

Comment:
If there's an error fetching search results, `r.data` will be undefined.
```suggestion
  $: searchResults = documents.then((r) => excludeDeleted($deleted, r.data));

  // filter out deleted documents that haven't been purged from search yet
  function excludeDeleted(
    deleted: Set<string>,
    searchResults?: DocumentResults, // optional params must come second
  ): DocumentResults {
    if (deleted.size === 0) return searchResults;

    const filtered = searchResults?.results.filter(
      (d) => !deleted.has(String(d.id)),
    ) ?? [];
```